tests/Unit/EventTickets/Actions/ConvertEventTicketsBlockToFieldsApiTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Give\Tests\Unit\EventTickets\Actions;
+
+use Exception;
+use Give\DonationForms\Models\DonationForm;
+use Give\EventTickets\Actions\ConvertEventTicketsBlockToFieldsApi;
+use Give\EventTickets\DataTransferObjects\EventTicketTypeData;
+use Give\EventTickets\Fields\EventTickets;
+use Give\EventTickets\Models\Event;
+use Give\EventTickets\Models\EventTicketType;
+use Give\Framework\Blocks\BlockModel;
+use Give\Framework\FieldsAPI\Exceptions\EmptyNameException;
+use Give\Tests\TestCase;
+use Give\Tests\TestTraits\RefreshDatabase;
+
+/**
+ * @unreleased
+ */
+class ConvertEventTicketsBlockToFieldsApiTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /**
+     * @unreleased
+     * @throws EmptyNameException
+     * @throws Exception
+     */
+    public function testBlockToFieldConversionMatchesAttributes(): void
+    {
+        $event = Event::factory()->create();
+        $ticketType = EventTicketType::factory()->create(['eventId' => $event->id]);
```

Comment:
The `EventTicketType` factory already creates a corresponding event, so creating and associating an event is not necessary.

Reply:
I need an `Event` object for my expectations. Creating one and associating it with the `EventTicketType` is quite similar to creating the `EventTicketType` first and then getting the `Event` object from it.

It's literally this change bellow.
```diff
- $event = Event::factory()->create();
$ticketType = EventTicketType::factory()->create(['eventId' => $event->id]);
+ $event = $ticketType->event()->get();
```

In fact, the proposed change would even add one more database query to the request.

Reply:
My point is that an additional, unintended `Event` is being created by this test setup. I'm lest concerned with adding an additional query than I am building a test environment with an unexpected extra event, which leads to troubleshooting problems in future maintenance.

Reply:
Changed on 318218bf8cb4a15be0d8c85cd80f64ba5050355f